Clarify "idle timeout" (#125)
quicwg/ops-drafts: draft-ietf-quic-applicability.md

## Session resumption versus Keep-alive {#resumption-v-keepalive}

Because QUIC is encapsulated in UDP, applications using QUIC must deal with
short network idle timeouts. Deployed stateful middleboxes will generally
establish state for UDP flows on the first packet state, and keep state for

> much shorter idle periods than for TCP. According to a 2010 study

Reviewer: Do we have any evidence that this recommendation is followed?  It seems safer to avoid mention of specifics for TCP, especially since they aren't really relevant to the current discussion.

> much shorter idle periods than for TCP. According to a 2010 study

Reviewer 2: I believe there's strong evidence it's not followed.  I believe I discussed this with @MikeBishop at some point, but I don't know if there's any relevant public data?

> much shorter idle periods than for TCP. According to a 2010 study

Author: Taking this suggestion but editing it a bit to reflect state of the literature AIUI.

> much shorter idle periods than for TCP. According to a 2010 study

Reviewer 3: Based on my vague memories of a study done at Microsoft several years ago, a 30-minute idle period on a TCP connection was tolerated by most firewalls, but beyond that you start finding your connection silently closed.

({{Hatonen10}}), UDP applications can assume that any NAT binding or other

> state entry will be expired after just thirty seconds of inactivity.

Reviewer 2: 'will' seems a bit strong.  Suggestion below

```suggestion
state entry can expire after just thirty seconds of inactivity.
```

> state entry will be expired after just thirty seconds of inactivity.

Reviewer: This is a fine suggestion, but I worry that this is starting to load up on text that is duplicative of the main spec.

> state entry will be expired after just thirty seconds of inactivity.

Author: In this case, it's more illustrative than duplicative (establishing context for UDP keepalive and why we make the recommendations we do in the doc), so I'd suggest adding the ref (and have done in another commit)

# Connection closure

> QUIC connections are closed either by expiration of an idle timeout, set by
> connection parameters, or by an

Reviewer 2: ```suggestion
QUIC connections are closed either by expiration of an idle timeout, as
negotiated by transport parameters, or by an
```

Reviewer: ```suggestion
QUIC connections are closed either by expiration of an idle timeout, as
determined by transport parameters, or by an
```
